examples/json/README.md

```diff
@@ -0,0 +1,164 @@
+# JSON example
+
+This example demonstrates writing a function that accepts a JSON request and
+sends a JSON response to the client.
+
+The basic shape of the function handler looks like this:
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
+}
+```
+
+The client will send a JSON document using the HTTP POST method. Here's an
+example request:
+
+```json
+{
+  "name": "World"
+}
+
+```
+
+The function will send a JSON document as the response. Here's an example
+response after receiving the above request:
+
+```json
+{
+  "salutation": "Hello",
+  "name": "World"
+}
+```
+
+The function handler actually receives the JSON request parsed as a map by the
+Functions Framework. To access the expected `name` field of the request
+document (and provide a default value in case the `name` field is empty), the
+code looks like this:
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
+  final name = request['name'] as String ?? 'World';
+}
+```
+
+Finally, the function creates an object returns it. The Functions Framework will
+take this and attempt "do the right" thing. In this case, the function is typed
+to return a `FutureOr<GreetingResponse>`, which has a `toJson()` method, so the
+framework will invoke this, then set the response body to the stringified result
+and set the response header (`content-type`) to `application/json`).
+
+```dart
+@CloudFunction()
+FutureOr<GreetingResponse> function(Map<String, dynamic> request) {
```

Comment:
Since this is not async, you can just return `GreetingResponse`

Reply:
True and yet 99% of the time there's going to be I/O and you're going to want a future. But you're right; when we have an example with some I/O, we can show that properly. Wondering if I should call it out here though... 🤔 Whaddaya think?

Reply:
Grateful to the linter for finding my link mistakes though! Hang on a moment.

Reply:
Alrighty then, fixed.